packages/deploy/src/stateTransform.ts

```diff
@@ -30,6 +31,24 @@ function stringifyJobBody(body: SpecJobBody): string {
   }
 }
 
+function transformSpecKafkaHost(hosts: string[]): StateKafkaHost[] {
+  function isValidHost(value: string): boolean {
+    const regex = /^([a-zA-Z0-9.-]+):(\d+)$/;
```

Comment:
I'm a little confused by `.` and `-` at the end of the host regex here? 

`.` means any character, so it kind of means the first pad means "match any character" and renders the rest obsolete.

I'm not sure why we need to support `-`

Reply:
Another pattern  I see for this sort of thing is to use the [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) object to parse a url string, and then test for `host` and `port` key.

I'm not convinced this approach is better than regex, but it's definitely better than a _bad_ regex :sweat_smile:

Reply:
Huh and maybe a dumb question - I'm nitpicking and anyway I don't know much about kafka... but does it _definitely_  need a port in the URL? It can't run off a default port?

Reply:
Yeah, it is AI generated 😅 . I did a couple of tests and it was able to capture all the bad cases I could think of.

I tried using the `URL` object but it proved troublesome because the protocol is missing from the required input.  `new URL('localhost:9092')` parses `localhost` as the `protocol`.  And yes, you MUST specify the port when connecting to a kafka cluster

Reply:
Let me see if I can simplify this. Maybe we don't even need a regex, the intention is to make sure that there is a `hostname:port_number` in the string. AI decided to take it a notch higher by trying to ensure that the hostname is a valid ip address